README.md

```diff
@@ -1,3 +1,141 @@
+```
+
+2. Decontextualize
+
+To decontextualize a snippet using some context, you can pass both the snippet and context to the decontextualization function.
+```python
+from qa_decontext import decontextualize
+
+context = """\
+Data collection. Subreddits are sub-communities on Reddit oriented around specific interests or topics, such as technology or politics. Sampling from Reddit as a whole would bias the model towards the most commonly discussed content. But by sampling posts from individual subreddits, we can control the kinds of posts we use to train our model. To collect a diverse training dataset, we have randomly s
+ampled 1000 posts each from the subreddits politics, business, science, and AskReddit, and 1000 additional posts from the Reddit frontpage. All posts in our sample appeared between January 2007 and March 2015, and to control for length effects, contain between 300 and 400 characters. This results in a total training dataset of 5000 posts.
+
+We compare the predictions of logistic regression models based on unigram bag-of-words features (BOW), sentiment signals (SENT), the linguistic features from our earlier analyses (LING), and combinations of these features. BOW and SENT provide baselines for the task. We compute BOW features using term frequency-inverse document frequency (TF-IDF) and category-based features by normalizing counts for each category by the number of words in each document. The BOW classifiers are trained with regularization (L2 penalties of 1.5).
+
+We now apply our dogmatism classifier to a larger dataset of posts, examining how dogmatic language shapes the Reddit community. Concretely, we apply the BOW+LING model trained on the full Reddit dataset to millions of new unannotated posts, labeling these posts with a probability of dogmatism according to the classifier (0=non-dogmatic, 1=dogmatic). We then use these dogmatism annotations to address four research questions."""
+
+snippet = "Concretely, we apply the BOW+LING model trained on the full Reddit dataset to millions of new unannotated posts, labeling these posts with a probability of dogmatism according to the classifier (0=non-dogmatic, 1=dogmatic)."
+
+decontextualize(snippet, context)
+> "[REF0] apply the BOW+LING [bag-of-words and linguistic features] model trained on the full Reddit dataset [different subreddit representing different topics, such as politics, business, science and other other posts in the Reddit home page] to millions of new unannotated posts, labeling these posts with a probability of dogmatism according to the classifier (0=non-dogmatic, 1=dogmatic)."
+```
+
+Often times, you want to use a whole paper or a list of papers as context. You can specify the papers as paths to `json` files with a specific format or as S2ORC IDs.
+```python
+paper_1 = "path/to/paper.json"
+paper_2 = "path/to/cited_paper.json"
+
+decontextualize(snippet, paper_1)
+decontextualize(snippet, [paper_1, paper_2])
+```
+
+Below is the `json` file format:
```

Comment:
Put this in a dataclass (instead of a path to a file).

Use Pydantic for everything. Use version <2.0.0.